c2corg_ui/static/js/imageuploader.js

```diff
@@ -0,0 +1,347 @@
+goog.provide('app.ImageUploaderController');
+goog.provide('app.imageUploaderDirective');
+goog.provide('app.ImageUploaderModalController')
+goog.require('app');
+
+
+/**
+ * This directive is used to display a drag-drop area for images to upload.
+ * @return {angular.Directive} The directive specs.
+ */
+app.imageUploaderDirective = function() {
+  return {
+    restrict: 'A',
+    controller: 'AppImageUploaderController',
+    controllerAs: 'uplCtrl',
+    bindToController: {'activities' : '=', 'categories': '=', 'types': '='},
+    templateUrl: '/static/partials/imageuploader.html',
+    link: function(scope, el, attrs, ctrl) {
+      // dropdown-menus can overlap the modal and it's impossible to fix it with overflow-y: srcoll + overflow-x: visible.
+      el.on('click', '.dropdown-toggle', function() {
+        ctrl.repositionMenu_(this);
+      });
+    }
+  };
+};
+
+app.module.directive('appImageUploader', app.imageUploaderDirective);
+
+
+/**
+ * @param {!angular.Scope} $scope Scope.
+ * @param {Object} $uibModal modal from angular bootstrap
+ * @param {angular.$compile} $compile Angular compile service.
+ * @param {angular.$q} $q promise
+ * @param {app.Api} appApi Api service.
+ * @param {app.Alerts} appAlerts
+ * @param {Object} Upload ng-file-upload Upload service.
+ * @constructor
+ * @export
+ * @ngInject
+ */
+app.ImageUploaderController = function($scope, Upload, $uibModal, $compile, $q, appAlerts, appApi) {
+
+  /**
+   * @type {Object} angular bootstrap modal
+   * @private
+   */
+  this.modal_ = $uibModal;
+
+  /**
+   * @type {app.Api}
+   * @private
+   */
+  this.api_ = appApi;
+
+
+  /**
+   * @type {app.Alerts}
+   * @private
+   */
+  this.alerts_ = appAlerts;
+
+
+  /**
+   * @type {angular.$compile}
+   * @private
+   */
+  this.compile_ = $compile;
+
+
+  /**
+   * @type {angular.$q}
+   * @private
+   */
+  this.q_ = $q;
+
+
+  /**
+   * @private
+   */
+  this.Upload_ = Upload;
+
+
+  /**
+   * @type {Array.<Object>}
+   * @export
+   */
+  this.uploading = [];
+
+
+  /**
+   * @type {Array.<Object>}
+   * @export
+   */
+  this.promises = [];
+
+
+  /**
+   * @type {Array.<Object>}
+   * @export
+   */
+  this.files = [];
+
+
+  /**
+   * @type {!angular.Scope}
+   * @private
+   */
+  this.scope_ = $scope;
+
+  this.scope_['activities'] = this['activities'];
```

Comment:
@gberaudo would probably say that you can use "dot" notation instead of "brackets" (at least for ``this.activities`` etc.).